### ISSUE TITLE
Eats newlines

### ISSUE THREAD

**Issue by tv42** [cespare/sub#10]
Something went wrong with the fix for #9. It seems lines that don't match at all get their newline eaten.
I think it's because of

https://github.com/cespare/sub/blob/35e96630fb5229ae3796bba2f63815bc6976e21d/sub.go#L74

```
$ printf 'one\ntwo\n' >z && od -a z && ./sub 'one$' three z && od -a z
0000000   o   n   e  nl   t   w   o  nl
0000010
z
0000000   t   h   r   e   e  nl   t   w   o
0000011
$ printf 'one\ntwo\n' >z && od -a z && ./sub 'two$' three z && od -a z
0000000   o   n   e  nl   t   w   o  nl
0000010
z
0000000   o   n   e   t   h   r   e   e  nl
0000011
```

**Comment by cespare** [comment]
Whoops, sorry about that. The tests for this thing are not great...

Fix incoming.

**Comment by cespare** [comment]
Fixed, and I released v0.3.0.

**Comment by tv42** [comment]
Sweet, thanks!
